library/manageiq.py

```diff
@@ -64,14 +61,86 @@
 '''
 
 
-def update_required(client, module, providers_url, provider_id, hostname, port):
-    try:
-        result = client.get(providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
-        endpoint = result['endpoints'][0]
-        return False if (endpoint['hostname'] == hostname and endpoint['port'] == int(port)) else True
-    except Exception as e:
-        msg = "Failed to get provider data. Error: %s" % e
-        module.fail_json(msg=msg)
+class ManageIQ(object):
+    """
+        ManageIQ object to execute various operations in manageiq
+
+        url      - manageiq environment url
+        user     - the username in manageiq
+        password - the user password in manageiq
+    """
+    openshift_provider_type = 'ManageIQ::Providers::OpenshiftEnterprise::ContainerManager'
+
+    def __init__(self, module, url, user, password):
+        self.module        = module
+        self.url           = url
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.url, (self.user, self.password))
+        self.changed       = False
+        self.msg           = ''
+        self.providers_url = self.url + '/providers'
+
+
+    def update_required(self, provider_id, hostname, port):
+        try:
+            result = self.client.get(self.providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
+            endpoint = result['endpoints'][0]
+            return False if (endpoint['hostname'] == hostname and endpoint['port'] == int(port)) else True
+        except Exception as e:
+            self.msg = "Failed to get provider data. Error: %s" % e
```

Comment:
@simon3z  it is standard to output a message as part of `module.exit_json` in case nothing failed (in case of failure the output is `module.fail_json`). the message is usually set in an inner method and stored as an attribute. 
I agree that in case of failure I could just do `msg =` and output `self.module.fail_json(msg=msg)` but I wanted to maintain consistency.